src/guest_memory.rs

```diff
@@ -336,6 +340,140 @@ pub trait GuestMemoryMut {
     }
 }
 
+/// GuestMemoryBackend is used to pass around the concrete GuestMemory types
+/// that should be used by a backend.  You can define a dummy struct that
+/// implements the trait, so that the GuestMemory clients can refer to the
+/// associated types in GuestMemoryBackend concisely.
+///
+/// ```
+/// # #[cfg(feature = "backend-mmap")]
+/// # use vm_memory::{GuestMemory, GuestMemoryBackend, GuestMemoryMmap};
+/// struct MmapBackend;
+/// # #[cfg(feature = "backend-mmap")]
+/// impl<'a> GuestMemoryBackend<'a> for MmapBackend {
+///     type Memory = GuestMemoryMmap;
+///     type AddressSpace = &'a GuestMemoryMmap;
+/// }
+/// ```
+
+pub trait GuestMemoryBackend<'a> {
```

Comment:
Hmm, why is this trait necessary/helpful in addition to `GuestAddressSpace`?  Is it mostly/mainly a solution to the discussion in #72? I noticed one of Paolo's comments saying some associated type(s) could not be moved in `GuestAddressSpace` so the backend remains separate; what was the desired outcome there?

Reply:
I have worked out a patch upon Paolo's patches to get rid of GuestMemoryBackend/GuestMemoryBackendMT. GuestMemory and GuestAddressSpace should be enough.
By that way we only add an extra layer GuestAddressSpace over GuestMemory.